XTS/WS-C/dev/src/com/arjuna/webservices11/wscoor/client/WSCOORClient.java

```diff
@@ -41,12 +42,27 @@
     private static synchronized ActivationService getActivationService()
     {
         if (activationService.get() == null) {
-            // activationService.set(new ActivationService(null, new QName("http://docs.oasis-open.org/ws-tx/wscoor/2006/06", "ActivationService")));
-            activationService.set(new ActivationService());
+            setActivationService();
         }
         return activationService.get();
     }
 
+    private static synchronized void setActivationService()
+    {
+        // activationService.set(new ActivationService(null, new QName("http://docs.oasis-open.org/ws-tx/wscoor/2006/06", "ActivationService")));
+        if (System.getSecurityManager() == null) {
+            activationService.set(new ActivationService());
+        } else {
+            AccessController.doPrivileged(new PrivilegedAction<Void>() {
+                @Override
+                public Void run() {
+                    activationService.set(new ActivationService());
```

Comment:
How about instead of using "new ActivationService()" I'll move that to the separate factory class. I could handle the security privileges there.
As a bonus I wouldn't need these extracted set methods and this would address your concern from above too.